packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -102,6 +152,43 @@ function buildCQLinks(pdResults: OutboundPatientDiscoveryResp[]): CQLink[] {
       oid: pd.gateway.oid,
       url,
       id: pd.gateway.id,
+      ...(pd.patientMatch ? { patientResource: pd.patientResource } : undefined),
```

Comment:
Can anyone explain why we save CQ links where we didn't match? Just "collect all data" vibes?

Reply:
I guess it's not clear to me based on the typing this is true (will look in the DB when I'm done testing).

This check:
`    
const id = pd.externalGatewayPatient?.id;
const system = pd.externalGatewayPatient?.system;
const url = pd.gateway.url;
if (!id || !system || !url) return [];
`

could pass with `patientMatch = false or null`.

externalGatewayPatient is defined as optional on `baseResponseSchema` and `gateway` is on `outboundPatientDiscoveryRespDefaultSchema`. Both of these schemas shared between `outboundPatientDiscoveryRespSuccessfulSchema` and `outboundPatientDiscoveryRespFaultSchema`... and there are checks within `createOrUpdateCQPatientData`.